internals/approval_defs.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import logging
+import requests
+
+
+ONE_LGTM = 'One LGTM'
+THREE_LGTM = 'Three LGTMs'
+API_OWNERS_URL = (
+    'https://source.chromium.org/chromium/chromium/src/+/'
+    # TODO(jrobbins): branch name will change soon
+    'master:third_party/blink/API_OWNERS')
+
+ApprovalFieldDef = collections.namedtuple(
+    'ApprovalField',
+    'name, description, field_id, rule, approvers')
+
+PrototypeApproval = ApprovalFieldDef(
+    'Intent to prototype approval',
+    'One API Owner must approve your intent',
+    1, ONE_LGTM, API_OWNERS_URL)
+
+ExperimentApproval = ApprovalFieldDef(
+    'Intent to experiment approval',
+    'One API Owner must approve your intent',
+    2, ONE_LGTM, API_OWNERS_URL)
+
+ShipApproval = ApprovalFieldDef(
+    'Intent to ship approval',
+    'Three API Owners must approve your intent',
+    3, THREE_LGTM, API_OWNERS_URL)
+
+APPROVAL_FIELDS_BY_ID = {
+    afd.field_id: afd
+    for afd in [PrototypeApproval, ExperimentApproval, ShipApproval]
+    }
+
+
+# XXX cache
+
+def fetch_owners(url):
+  """Load a list of email addresses from an OWNERS file."""
+  owners = []
+  response = requests.get(url)
+  if response.status_code != 200:
+    logging.error('Could not fetch %r', url)
+    logging.error('Got response %r', response)
+    raise ValueError('Could not get OWNERS file')
+
+  for line in response.iter_lines():
+    print('line is %r' % line)
+    if '#' in line:
+      line = line[:line.index('#')]
+    line = line.strip()
+    if '@' in line and '.' in line:
+      owners.append(line)
+
+  return owners
+
+
+def get_approvers(field_id):
+  """Return a list of email addresses of users allowed to approve."""
+  afd = APPROVAL_FIELDS_BY_ID[field_id]
+
+  if isinstance(afd.approvers, str):
+    owners = fetch_owners(afd.approvers)
+    return owners
```

Comment:
Did you mean to set afd.approvers here as well? If not, I don't see how it's ever not a string?

Reply:
That's correct.  In the only examples that we have right now for API Owners the value of afd.approvers will be a URL to fetch.  We keep that URL so that we can fetch it again periodically.    In the future, we may have some other approval types that just have a hard-coded list of approvers.  I'll add a comment to explain that.